data/result-exporter/swe-cne-exporter/src/main/java/com/farao_community/farao/data/swe_cne_exporter/SweRemedialActionSeriesCreator.java

```diff
@@ -127,17 +127,30 @@ public List<RemedialActionSeries> generateRaSeriesReference(Contingency continge
     }
 
     private RemedialActionSeries generateRaSeries(State state, RemedialActionSeriesCreationContext context, boolean onlyReference) {
-        RaoResult raoResult = cneHelper.getRaoResult();
-        Crac crac = cneHelper.getCrac();
-        List<RemedialAction<?>> usedRas = context.getCreatedIds().stream().sorted()
-            .map(crac::getRemedialAction)
-            .filter(ra -> raoResult.isActivatedDuringState(state, ra)).collect(Collectors.toList());
+        RaoResult raoResult = sweCneHelper.getRaoResult();
+        AngleMonitoringResult angleMonitoringResult = sweCneHelper.getAngleMonitoringResult();
+        Crac crac = sweCneHelper.getCrac();
+        List<RemedialAction<?>> usedRas = new ArrayList<>();
+        if (Objects.nonNull(raoResult)) {
+            usedRas.addAll(context.getCreatedIds().stream().sorted()
+                    .map(crac::getRemedialAction)
+                    .filter(ra -> raoResult.isActivatedDuringState(state, ra)).collect(Collectors.toList()));
+        }
+        if (Objects.nonNull(angleMonitoringResult) && !angleMonitoringResult.isDivergent()) {
+            usedRas.addAll(context.getCreatedIds().stream().sorted()
+                    .map(crac::getRemedialAction)
+                    .filter(ra -> angleMonitoringResult.getAppliedCras(state).stream().anyMatch(cra -> cra.getId().equals(ra.getId())))
+                    .collect(Collectors.toList()));
```

Comment:
same: instead of collecting then using addAll, you can use foreach(usedRas::add)